index.js

```diff
@@ -0,0 +1,104 @@
+import {
+    bookSection,
+    titleText,
+    authorText,
+    form,
+    listbtn,
+    addlistbtn,
+    contactbtn,
+    list,
+    newbooks,
+    contact,
+    date,
+} from "./modules/book.js";
+
+class Book {
+  constructor() {
+    this.bookLibrary = JSON.parse(localStorage.getItem("jsonLibrary")) || [];
+  }
+
+  insertHtml() {
+    bookSection.innerHTML = "";
+    this.bookLibrary.forEach((n) => {
+      bookSection.innerHTML += `<div class='book'>
+        <div class='bookDetail'>
+          <p>"${n.title}"</p>
+          <p>by</p>
+          <p>${n.author}</p>
+        </div>    
+          <button class='remove' type='button' onclick='bookList.remove("${n.id}")'>Remove</button>
+      </div>`;
+    });
+  }
+
+  addBook() {
+    const bookObject = {};
+    bookObject.title = titleText.value;
+    bookObject.author = authorText.value;
+    bookObject.id = (titleText.value + authorText.value).replace(/\s/g, "");
+    this.bookLibrary.push(bookObject);
+    localStorage.setItem("jsonLibrary", JSON.stringify(this.bookLibrary));
+    this.insertHtml();
+  }
+
+  remove(idParameter) {
+    this.bookLibrary = this.bookLibrary.filter(
+      (book) => book.id !== idParameter
+    );
+    localStorage.setItem("jsonLibrary", JSON.stringify(this.bookLibrary));
+    this.insertHtml();
+  }
```

Comment:
- [ ] This was carefully written but it seems to me that something is missing because it is unable to remove or delete books on request as shown in this [gif picture](https://www.webmobilefirst.com/en/screencasts/ivdjox4sNr/).

Reply:
- Refactored `Book` class to create `booksection` elements dynamically with JS to get more control over the elements fixing the remove bug issue.